### ISSUE TITLE
changing margin on last page

### ISSUE THREAD

**Issue by anarcat** [profound-labs/wallcalendar#15]
in https://github.com/profound-labs/wallcalendar/issues/4#issuecomment-433684222 you said extra content could be added at the end of the document to append pages to the calendar. however, my experience with this has been a little difficult, because normal methods to control layout on that page fail.

for example, the margins on the page are very wide, which makes it difficult to add much content at all. i tried to use the `geometry` package to fix that, but as soon as the package is included in the preamble, it throws all of the other pages out of whack. images are misaligned:

![misaligned image](https://user-images.githubusercocom/796623/49775375-fee55780-fcc5-11e8-8c99-b836d54bb09f.png)

... and the notes area is broken as well:

![notes overlapping with calendar](https://user-images.githubusercontent.com/796623/49775392-0efd3700-fcc6-11e8-9b31-775d731cbee4.png)

so that doesn't seem to be an option... what can i do to control layout in the last page? alternatives to geometry?

specifically, I would like to reduce margins to a minimum, or at least something that matches better the rest of the calendar...

thanks!

**Comment by anarcat** [comment]
FWIW, i tried:

```tex
\newpage

\setlength{\topmargin}{-40mm}
\setlength{\oddsidemargin}{-10mm}
\setlength{\evensidemargin}{5mm}

blablabla
```

but those also affect the previous page.

**Comment by gambhiro** [comment]
It is possible to change the page layout mid-document.

The `wallcalendar` package builds on `memoir`, so you have to use memoir's page layout mechanisms.

Refer to memoir's documentation, chapter `Laying out the page`.

Also note:

[Changing textblock and margin size mid-document with memoir](https://tex.stackexchange.com/questions/110201/changing-textblock-and-margin-size-mid-document-with-memoir)

For example, define a command in the preamble such as

```
\newcommand{\newMargins}{%
  \makeatletter%
  \settypeblocksize{172mm}{96mm}{*}%
  \setlrmargins{*}{25mm}{*}%
  \checkandfixthelayout%
  \ch@ngetext%
  \makeatother%
}
```

Then in the document body:

```
\clearpage
\newMargins

Lorem ipsum dolor sit amet, consectetuer adipiscing elit.  Donec hendrerit tempor tellus.  Donec pretium posuere tellus.  Proin quam nisl, tincidunt et, mattis eget, convallis nec, purus.  Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.  Nulla posuere.  Donec vitae dolor.  Nullam tristique diam non turpis.  Cras placerat accumsan nulla.  Nullam rutrum.  Nam vestibulum accumsan nisl.
```

**Comment by anarcat** [comment]
great, that's super helpful! i'll see if i can update the docs to reflect this, as it's not obvious from the outside, especially for a novice like me. ;)

`\setlrmargins` seems to work well - it reduces the margin on the left but expands the margin on the right!  I suspect the right margin may be due to the "marginal notes" (section 2.5) so I tried to add `\setmarginnotes{0mm}{0mm}{0mm}` to the `\newMargins` command, and that didn't seem to do anything. I also tried to provide another argument to `\setrlmargins` but that apparently doesn't make sense (it's "ambiguous" according to table 2.4). And `\settypeblocksize` causes weird problems - it looks like it shifts everything in a right column, with nothing in the left. So I settled with:

```
\newcommand{\newMargins}{%
  \makeatletter%
  \setlrmarginsandblock{10mm}{10mm}{*}%
  \checkandfixthelayout%
  \ch@ngetext%
  \makeatother%
}
```

I'm not too happy about the `10mm` values - I picked them from trial and error, but that will have to do for now. I'm also wondering if I should be doing this with columns instead which would make the text more readable and allow for better control... But at this point, I just want to finish the project. :)

Thanks so much for the help!
